src/libspark/keys.cpp

```diff
@@ -115,4 +134,47 @@ const GroupElement& Address::get_Q2() const {
 	return this->Q2;
 }
 
+std::string Address::GetHex() const {
+    const std::size_t size = 2* GroupElement::serialize_size + AES_BLOCKSIZE;
+    std::vector<unsigned char> buffer;
+    buffer.reserve(size);
+    buffer.resize(2* GroupElement::serialize_size);
+    unsigned char* ptr = buffer.data();
+    ptr = Q1.serialize(ptr);
+    Q2.serialize(ptr);
+    buffer.insert(buffer.end(), d.begin(), d.end());
+
+    std::stringstream ss;
+    ss << std::hex;
+    for (const auto b : buffer) {
+        ss << (b >> 4);
+        ss << (b & 0xF);
+    }
+
+    return ss.str();
+}
+
+void Address::SetHex(const std::string& str) {
```

Comment:
In the event that `str` is invalid and an exception is thrown, `this` will be trampled.

Reply:
What you mean trampled? Anyway you should handle all cases and catch exceptions.